src/go/rpk/pkg/kafka/client.go

```diff
@@ -171,6 +180,29 @@ func RetrySend(
 	return part, offset, err
 }
 
+func configureSASL(saramaConf *sarama.Config, rpConf *config.Config,
+) (*sarama.Config, error) {
+	rp := rpConf.Rpk
+	if rp.SCRAMPassword == "" || rp.SCRAMUser == "" && rp.SCRAMType == "" {
+		return saramaConf, nil
+	}
+
+	saramaConf.Net.SASL.Enable = true
+	saramaConf.Net.SASL.Handshake = true
+	saramaConf.Net.SASL.User = rp.SCRAMUser
+	saramaConf.Net.SASL.Password = rp.SCRAMPassword
+	saramaConf.Net.SASL.Mechanism = rp.SCRAMType
+	switch rp.SCRAMType {
+	case sarama.SASLTypeSCRAMSHA256:
+		saramaConf.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &XDGSCRAMClient{HashGeneratorFcn: sha256.New} }
+	case sarama.SASLTypeSCRAMSHA512:
+		saramaConf.Net.SASL.SCRAMClientGeneratorFunc = func() sarama.SCRAMClient { return &XDGSCRAMClient{HashGeneratorFcn: sha512.New} }
+	default:
+		return nil, fmt.Errorf("unrecongnized Salted Challenge Response Authentication Mechanism (SCRAM): %s", rp.SCRAMType)
```

Comment:
Worth pointing out the exact config path too (`rpk.scram.type`)

Reply:
Ok changed.